pkg/api/mlflow/dao/repositories/helpers.go

```diff
@@ -18,17 +19,33 @@ func makeSqlPlaceholders(numberInEachSet, numberOfSets int) string {
 
 // makeParamConflictPlaceholdersAndValues provides sql placeholders and concatenates
 // Key, Value, RunID from each input Param for use in sql values replacement
-func makeParamConflictPlaceholdersAndValues(params []models.Param) (string, []interface{}) {
-	// make place holders of 3 fields for each param
-	placeholders := makeSqlPlaceholders(3, len(params))
-	// values array is params * 3 in length since using 3 fields from each
-	valuesArray := make([]interface{}, len(params)*3)
+func makeParamConflictPlaceholdersAndValues(params []models.Param, dialector string) (string, []interface{}) {
+	var placeholders string
+	// make place holders of 5 fields for each param
+	if (sqlite.Dialector{}.Name() == dialector) {
+		placeholders = fmt.Sprintf("VALUES %s", makeSqlPlaceholders(5, len(params)))
+	} else {
+		set := "SELECT ?::text, ?::text, ?::int, ?::float, ?::text"
+		placeholders = strings.Repeat(set+"\nUNION ALL\n", len(params)-1) + set
+	}
+	// values array is params * 5 in length since using 5 fields from each
+	valuesArray := make([]interface{}, len(params)*5)
 	index := 0
 	for _, param := range params {
+		var valueInt, valueFloat, valueStr any
+		if param.ValueInt != nil {
+			valueInt = *param.ValueInt
+		} else if param.ValueFloat != nil {
+			valueFloat = *param.ValueFloat
+		} else if param.ValueStr != nil {
+			valueStr = *param.ValueStr
+		}
```

Comment:
I think `GetValueAsString` etc, implies a conversion, which we do not want here

Reply:
i think we need to return nil or the dereferenced value in each case, but without causing nil dereference error. So somewhere we have to check `if != nil` before deferencing. I think wherever we put it will have "scary if statements"

Reply:
i simplified this by removing the intermediary vars, but I think functions will confuse the issue -- we are turning a pointer into dereferenced value or nil, so there has to be a nil check, but a function would either return another pointer or some sort of Optional like sql.NullInt64, etc, so the `valid` would still need checking with scary `if`.